### ISSUE TITLE
Windows (plain/WSL2) performance when running dev (without --host)

### ISSUE THREAD

**Issue by deStrO** [nuxt/cli#109]
### Environment

- Operating System: Linux
- Node Version:     v18.16.0
- Nuxt Version:     3.7.0
- CLI Version:      3.7.2
- Nitro Version:    2.6.2
- Package Manager:  yarn@1.22.19
- Builder:          -
- User Config:      experimental, modules, swiper, css, gtm, components, auth, rootDir, ssr, app, tailwindcss, build, ignore, i18n, runtimeConfig
- Runtime Modules:  @nuxtjs/tailwindcss@6.6.7, nuxt-swiper@1.2.2, @nuxtjs/i18n@8.0.0-rc.3, @nuxt/devtools@0.4.2, @sidebase/nuxt-auth@0.6.0-beta.4, @zadigetvoltaire/nuxt-gtm@0.0.13
- Build Modules:    -

### Reproduction

Just run yarn dev without --host in a WSL2 container (Debian)

### Describe the bug

All of my pages are loading slow, like components per components.
Adding --host to yarn dev command solve the issue, but don't have any clue why.

![image](https://github.com/nuxt/nuxt/assets/519112/aac10b33-135e-45d9-ae20-604a8fc4004f)

### Additional context

Upgraded a project from 3.4.3 to 3.7.0, and the loading of the project was slow (super slow).

**Comment by pi0** [comment]
Thanks for reporting issue @deStrO and confirming anyhost with `--host` fixes WSL @jensk 👍🏼 

I will have to double check on my local maching. I think we can detect wsl with std-env and apply this fix if couldn't find the root cause of the problem on WSL...

**Comment by danielroe** [comment]
That seems like a good idea to me.

**Comment by lihbr** [comment]
Just documenting I'm able to reproduce on **plain Windows** (no WSL). `--host` also solves the issue somehow 🤔

```bash
npx nuxi@latest init

# stalling requests
npm run dev

# no stalling requests
npm run dev -- --host
```


https://github.com/nuxt/cli/assets/25330882/cfb9dacb-8a5d-47d5-a23e-1b208a92cb54

cc @pi0

**Comment by pi0** [comment]
@lihbr Thanks for confirming. While your windows Machine is warm (just kidding it is windows :p), can you please also try with `--host 127.0.0.1` as well if it makes any change or same effect of solving perf issue?

**Comment by deStrO** [comment]
@pi0 same issue, stalled requests on my side.

**Comment by pi0** [comment]
@deStrO Can you please also try with `--host localhost` and `--host ::1`? 🙏🏼 (this is important to avoid `--host` and have it as last resort to avoid exposing nuxt server to local network by default in windows. however there is firewall by default it is less safe)

**Comment by pi0** [comment]
Okay, so we have two confirmations now:

- @lihbr Both `::1` and `127.0.0.1` work
- @deStrO Only `::1` works and `127.0.0.1` seems to be slow

**Comment by lihbr** [comment]
Not sure @deStrO confirmed anything with `127.0.0.1`(?) but maybe I read it wrong.

Tried again to be sure, I can confirm `npm run dev -- --host 127.0.0.1` works on my end (I don't have any funny setup for that to work more like linux afaik also)

**Comment by deStrO** [comment]
It's slow with --host 127.0.0.1 (WSL2 / Debian)

**Comment by pi0** [comment]
I am thinking now to:

- Use `127.0.0.1` when running in actual windows host
- Use `::1` or `` (anyhost same as before) when running within WSL2 (still need another confirmation for WSL2 behavior).

**Comment by Dodobibi** [comment]
https://github.com/nuxt/nuxt/discussions/22823#discussioncomment-6840605

**Comment by pi0** [comment]
Fix is released in [v3.7.3](https://github.com/nuxt/cli/releases/tag/v3.7.3).

**Comment by pi0** [comment]
Nuxi [3.9.0](https://github.com/nuxt/cli/releases/tag/v3.9.0) is out to fix a couple of relevant issues! 

Please report if you are still experiencing slow dev server issues to https://github.com/nuxt/cli/issues/209 🙏🏼
